tcf_website/views/review.py

```diff
@@ -63,6 +63,17 @@ def new_review(request):
             instance.hours_per_week = \
                 instance.amount_reading + instance.amount_writing + \
                 instance.amount_group + instance.amount_homework
+
+            # Verify that user has not already submitted a review for this
+            # specific course for the same semester and instructor before
+            for review in request.user.reviews():
+                if review.__str__() == instance.__str__() and review.semester == instance.semester:
```

Comment:
This doesn't seem like a very good way to compare Reviews... for example, I can review CS 2110 twice as long as I put a different instructor. 

![image](https://user-images.githubusercontent.com/55100084/108609653-5353e980-739d-11eb-9b68-ff2a3e4d5812.png)

I would do something like `review.course == instance.course` — disclaimer: haven't tested that, so that might not work. The idea is that we should be comparing data directly instead of a string representation.

Reply:
One exception I can think of: during Covid, some profs have been co-teaching and alternating lectures (Wu/Brunelle for Algo, Sherriff/McBurney for 3240), so in that sense people _might_ want to review separately? But I highly doubt that people would go to the extent to specifically do this.

Reply:
Yeah I considered being able to review for the exact same course but different instructor and thought that was generally fine, in case you interacted with that professor in some other capacity especially now with co-teaching as you mentioned. I didn’t think writing reviews for identical courses but different instructors was problematic so I left it out but I’m open to changing this. I doubt people would bother repeating reviews for different instructors anyway.

Reply:
Will test your suggestion though along with other methods though

Reply:
Barring Alex's corner cases, I would use something like this (might not be accurate Django code):
```python
reviews_on_same_class = request.user.review_set.filter(
    course=instance.course,
    instructor=instance.instructor,
    semester=instance.semester,
)
if reviews_on_same_class.exists():
    ...
```

Reply:
As of right now I've made it specific to the same course + instructor + semester (as @Bartleby2718 suggested), but I can change it to prevent reviewing the overall course again? (i.e. with any instructor/semester; so only filter for course=instance.course)